File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {
    let ws=0;
    for(let i=0;i<k;i++){
        ws+=nums[i];
    }
    let ms=ws;
    for(let i=k;i<nums.length;i++){
        ws+=nums[i]-nums[i-k];
        ms=Math.max(ms,ws);
    }
    return ms/k;
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int sum=0;
        int msum;
        for(int i=0;i<k;i++){
            sum+=nums[i];
        }
        msum=sum;
        for(int i=k;i<nums.size();i++){
            sum-=nums[i-k];
            sum+=nums[i];
            msum=max(msum,sum);
        }
        return (double)msum/k;

    }
};